### PR TITLE
Updating HydroDispatch

### PR DESCRIPTION
**Pull request by sourabhdalvi**
- Deleting HydroStorage
- Adding `storagecapacity` & `initial_storage` fields  to HydroDispatch struct
- Adding  `storage::Union{DataFrames.DataFrame,Nothing}` field to PowerSystemTableData to read in `storage.csv` for Hydro storage related fields

For the test to pass it require the sd/hydro_update branch on PowerSystemsTestData repo to be merged into master

### CODE REVIEW

File: src/parsers/pm2ps_parser.jl [hunk 1]
```diff
@@ -211,6 +211,8 @@ function make_hydro_gen(gen_name, d, bus)
         reactivepower = d["qg"],
         tech = tech,
         op_cost = curtailcost,
+        storagecapacity = 0.0, #TODO: Implement better Solution for this
```
Comment (jd-lara) [inline src/parsers/pm2ps_parser.jl]: storage_capacity

File: src/parsers/power_system_table_data.jl [hunk 2]
```diff
@@ -13,6 +13,7 @@ struct PowerSystemTableData
     gen::Union{DataFrames.DataFrame,Nothing}
     load::Union{DataFrames.DataFrame,Nothing}
     services::Union{DataFrames.DataFrame,Nothing}
+    storage::Union{DataFrames.DataFrame,Nothing}
```
Comment (jd-lara) [inline src/parsers/power_system_table_data.jl]: Is this inflow?

File: src/models/generated/HydroDispatch.jl [hunk 3]
```diff
@@ -10,6 +10,8 @@ This file is auto-generated. Do not edit.
         reactivepower::Float64
         tech::TechHydro
         op_cost::TwoPartCost
+        storagecapacity::Float64
```
Comment (jd-lara) [inline src/models/generated/HydroDispatch.jl]: we might need to add an inflow field as well here.

File: src/parsers/power_system_table_data.jl (hunk 2 above)
Comment (sourabhdalvi) [reply src/parsers/power_system_table_data.jl]: Thats for the `storage.csv` file in data/RTS_GMLC which currently holds :
- Max Volume GWh,
- Initial Volume GWh,
- Start Energy  GWh,
I think Inflow would be in  the data/forecasts

File: src/models/generated/HydroDispatch.jl (hunk 3 above)
Comment (sourabhdalvi) [reply src/models/generated/HydroDispatch.jl]: ok, I can do that. Should `storage_capacity` and `inflow`fields be in the TechHydro ?

**Comment by sourabhdalvi** [comment]
> We need to add the inflow field and also include in tech_hydro a conversion function from flow to power. We need to make sure the units are all consistent here. For instance, flow in m^3/s and storage en m^3.

I and @claytonpbarrows  talked about this yesterday and it seems like most of the inflow information that is used/available at NREL is in terms of energy. So Clayton suggested we keep everything in energy units.  So `storage_capacity` and `inflow` would both in terms of energy and that way its easier to update the parameter for inflow in PSI to keep track of stored energy over the length of the simulation.

**Comment by claytonpbarrows** [comment]
> > We need to add the inflow field and also include in tech_hydro a conversion function from flow to power. We need to make sure the units are all consistent here. For instance, flow in m^3/s and storage en m^3.
> 
> I and @claytonpbarrows talked about this yesterday and it seems like most of the inflow information that is used/available at NREL is in terms of energy. So Clayton suggested we keep everything in energy units. So `storage_capacity` and `inflow` would both in terms of energy and that way its easier to update the parameter for inflow in PSI to keep track of stored energy over the length of the simulation.

Yes. I also agree that for some formulations/implementations it would be useful to keep track of the water itself (inflow, level, etc.). But, I think we can add that later.

**Pull request merged**